bin/crew

```diff
@@ -1125,10 +1125,44 @@ def install_package(pkgdir)
     FileUtils.mv 'dlist', CREW_META_PATH + @pkg.name + '.directorylist', verbose: @fileutils_verbose
     FileUtils.mv 'filelist', CREW_META_PATH + @pkg.name + '.filelist', verbose: @fileutils_verbose
 
+    @brokensymlinks = nil
+    @brokensymlinks = %x[find . -type l -exec test ! -e {} \\; -print].chomp
+    unless @brokensymlinks.nil? or @brokensymlinks.empty?
+      puts "There are broken symlinks. Will try to fix.".orange if @opt_verbose
+      @brokensymlinks.each_line do |fixlink|
+        @brokentarget = nil
+        @fixedtarget = nil
+        @brokentarget = %x[readlink -n #{fixlink}]
+        puts "Attempting fix of: #{fixlink.delete_prefix('.')} -> #{@brokentarget}".orange if @opt_verbose
+        @fixedtarget = @brokentarget.delete_prefix(CREW_DEST_DIR)
+        # If no changes were made, don't replace symlink
+        unless @fixedtarget == @brokentarget
+          # This doesn't work. Why?
+          FileUtils.ln_sf @fixedtarget,pkgdir + fixlink.delete_prefix('.')
+          # This works.
+          system "ln -sf #{@fixedtarget} #{pkgdir + fixlink.delete_prefix('.')}"
```

Comment:
That does not work either.

Reply:
What is the error message?  Maybe assign `pkgdir + fixlink.delete_prefix('.')` to a variable first?

Reply:
There is no error message.
This also does not do anything:
```
          @fixlink_loc = pkgdir + fixlink.delete_prefix('.')
          FileUtils.ln_sf @fixedtarget,@fixlink_loc
```

Reply:
What is the output of `puts @fixlink_loc`?  Does it make sense?

Reply:
output from:
```
    unless @brokensymlinks.nil? or @brokensymlinks.empty?
      puts "There are broken symlinks. Will try to fix.".orange if @opt_verbose
      @brokensymlinks.each_line do |fixlink|
        @brokentarget = nil
        @fixedtarget = nil
        @brokentarget = %x[readlink -n #{fixlink}]
        puts "Attempting fix of: #{fixlink.delete_prefix('.')} -> #{@brokentarget}".orange if @opt_verbose
        @fixedtarget = @brokentarget.delete_prefix(CREW_DEST_DIR)
        # If no changes were made, don't replace symlink
        unless @fixedtarget == @brokentarget
          # This doesn't work. Why?
          #FileUtils.ln_sf @fixedtarget,pkgdir + fixlink.delete_prefix('.')
          @fixlink_loc = pkgdir + fixlink.delete_prefix('.')
          puts @fixlink_loc
          puts "before...".orange
          system "ls -aFl #{@fixlink_loc}"
          FileUtils.ln_sf @fixedtarget,@fixlink_loc
          puts "after...".orange
          system "ls -aFl #{@fixlink_loc}"
          # This works.
          #system "ln -sf #{@fixedtarget} #{pkgdir + fixlink.delete_prefix('.')}"
          puts "Fixed: #{@fixedtarget} -> #{fixlink.delete_prefix('.')}".orange if @opt_verbose
        end
      end
    end
```

![image](https://user-images.githubusercontent.com/1096701/145890228-8aa70002-bcad-49d8-934d-c2cdc3fabb8a.png)

What's interesting is that it works for the last one, but not for the earlier ones...

Reply:
try changing the shebang on chromebrew's executable to `/usr/bin/env -S ruby -d`

Reply:
@saltedcoffii I see this:
```
There are broken symlinks. Will try to fix.
Attempting fix of: /usr/local/lib64/clang/13.0.1/include/profile/InstrProfData.inc
 -> /usr/local/tmp/crew/dest/usr/local/include/llvm/ProfileData/InstrProfData.inc
/usr/local/tmp/crew/llvm.20211214173343.dir/usr/local/lib64/clang/13.0.1/include/profile/InstrProfData.inc
before...
lrwxrwxrwx 1 chronos chronos 77 Dec  5 23:38 /usr/local/tmp/crew/llvm.20211214173343.dir/usr/local/lib64/clang/13.0.1/include/profile/InstrProfData.inc -> /usr/local/tmp/crew/dest/usr/local/include/llvm/ProfileData/InstrProfData.inc
Exception `Errno::ENOENT' at /usr/local/lib64/ruby/3.0.0/fileutils.rb:1466 - No such file or directory @ apply2files - /usr/local/tmp/crew/llvm.20211214173343.dir/usr/local/lib64/clang/13.0.1/include/profile/InstrProfData.inc

after...
lrwxrwxrwx 1 chronos chronos 77 Dec  5 23:38 /usr/local/tmp/crew/llvm.20211214173343.dir/usr/local/lib64/clang/13.0.1/include/profile/InstrProfData.inc -> /usr/local/tmp/crew/dest/usr/local/include/llvm/ProfileData/InstrProfData.inc
Fixed: /usr/local/include/llvm/ProfileData/InstrProfData.inc -> /usr/local/lib64/clang/13.0.1/include/profile/InstrProfData.inc

Attempting fix of: /usr/local/libexec/intercept-cc
 -> /usr/local/tmp/crew/dest/usr/local/libexec/intercept-c++
/usr/local/tmp/crew/llvm.20211214173343.dir/usr/local/libexec/intercept-cc
before...
lrwxrwxrwx 1 chronos chronos 56 Dec  5 23:38 /usr/local/tmp/crew/llvm.20211214173343.dir/usr/local/libexec/intercept-cc -> /usr/local/tmp/crew/dest/usr/local/libexec/intercept-c++
Exception `Errno::ENOENT' at /usr/local/lib64/ruby/3.0.0/fileutils.rb:1466 - No such file or directory @ apply2files - /usr/local/tmp/crew/llvm.20211214173343.dir/usr/local/libexec/intercept-cc

after...
lrwxrwxrwx 1 chronos chronos 56 Dec  5 23:38 /usr/local/tmp/crew/llvm.20211214173343.dir/usr/local/libexec/intercept-cc -> /usr/local/tmp/crew/dest/usr/local/libexec/intercept-c++
Fixed: /usr/local/libexec/intercept-c++ -> /usr/local/libexec/intercept-cc

Attempting fix of: /usr/local/lib/libscanbuild/resources/scanview.css -> /usr/local/tmp/crew/dest/usr/local/share/scan-build/scanview.css
/usr/local/tmp/crew/llvm.20211214173343.dir/usr/local/lib/libscanbuild/resources/scanview.css
before...
lrwxrwxrwx 1 chronos chronos 64 Dec  5 23:38 /usr/local/tmp/crew/llvm.20211214173343.dir/usr/local/lib/libscanbuild/resources/scanview.css -> /usr/local/tmp/crew/dest/usr/local/share/scan-build/scanview.css
after...
lrwxrwxrwx 1 chronos chronos 40 Dec 14 17:33 /usr/local/tmp/crew/llvm.20211214173343.dir/usr/local/lib/libscanbuild/resources/scanview.css -> /usr/local/share/scan-build/scanview.css
Fixed: /usr/local/share/scan-build/scanview.css -> /usr/local/lib/libscanbuild/resources/scanview.css
```

So I went further and tried this:

```
@brokensymlinks = nil
    @brokensymlinks = %x[find . -type l -exec test ! -e {} \\; -print].chomp
    unless @brokensymlinks.nil? or @brokensymlinks.empty?
      puts "There are broken symlinks. Will try to fix.".orange if @opt_verbose
      @brokensymlinks.each_line do |fixlink|
        @brokentarget = nil
        @fixedtarget = nil
        @brokentarget = %x[readlink -n #{fixlink}]
        puts "Attempting fix of: #{fixlink.delete_prefix('.')} -> #{@brokentarget}".orange if @opt_verbose
        @fixedtarget = @brokentarget.delete_prefix(CREW_DEST_DIR)
        # If no changes were made, don't replace symlink
        unless @fixedtarget == @brokentarget
          # This doesn't work. Why?
          #FileUtils.ln_sf @fixedtarget,pkgdir + fixlink.delete_prefix('.')
          @fixlink_loc = pkgdir + fixlink.delete_prefix('.')
          puts @fixlink_loc
          puts "before...".orange
          system "ls -aFl #{@fixlink_loc}"
          puts "rm #{@fixlink_loc}".orange if @opt_verbose
          begin
            FileUtils.rm @fixlink_loc
          rescue
            system "ls -aFl #{@fixlink_loc}"
          end
          FileUtils.ln_sf @fixedtarget,@fixlink_loc
          puts "after...".orange
          system "ls -aFl #{@fixlink_loc}"
          # This works.
          #system "ln -sf #{@fixedtarget} #{pkgdir + fixlink.delete_prefix('.')}"
          puts "Fixed: #{@fixedtarget} -> #{fixlink.delete_prefix('.')}".orange if @opt_verbose
        end
      end
    end
```
Leading to:
```
There are broken symlinks. Will try to fix.
Attempting fix of: /usr/local/lib64/clang/13.0.1/include/profile/InstrProfData.inc
 -> /usr/local/tmp/crew/dest/usr/local/include/llvm/ProfileData/InstrProfData.inc
/usr/local/tmp/crew/llvm.20211214175054.dir/usr/local/lib64/clang/13.0.1/include/profile/InstrProfData.inc
before...
lrwxrwxrwx 1 chronos chronos 77 Dec  5 23:38 /usr/local/tmp/crew/llvm.20211214175054.dir/usr/local/lib64/clang/13.0.1/include/profile/InstrProfData.inc -> /usr/local/tmp/crew/dest/usr/local/include/llvm/ProfileData/InstrProfData.inc
rm /usr/local/tmp/crew/llvm.20211214175054.dir/usr/local/lib64/clang/13.0.1/include/profile/InstrProfData.inc

Exception `Errno::ENOENT' at /usr/local/lib64/ruby/3.0.0/fileutils.rb:1466 - No such file or directory @ apply2files - /usr/local/tmp/crew/llvm.20211214175054.dir/usr/local/lib64/clang/13.0.1/include/profile/InstrProfData.inc

Exception `Errno::ENOENT' at /usr/local/lib64/ruby/3.0.0/fileutils.rb:800 - No such file or directory @ apply2files - /usr/local/tmp/crew/llvm.20211214175054.dir/usr/local/lib64/clang/13.0.1/include/profile/InstrProfData.inc

lrwxrwxrwx 1 chronos chronos 77 Dec  5 23:38 /usr/local/tmp/crew/llvm.20211214175054.dir/usr/local/lib64/clang/13.0.1/include/profile/InstrProfData.inc -> /usr/local/tmp/crew/dest/usr/local/include/llvm/ProfileData/InstrProfData.inc
Exception `Errno::ENOENT' at /usr/local/lib64/ruby/3.0.0/fileutils.rb:1466 - No such file or directory @ apply2files - /usr/local/tmp/crew/llvm.20211214175054.dir/usr/local/lib64/clang/13.0.1/include/profile/InstrProfData.inc

after...
lrwxrwxrwx 1 chronos chronos 77 Dec  5 23:38 /usr/local/tmp/crew/llvm.20211214175054.dir/usr/local/lib64/clang/13.0.1/include/profile/InstrProfData.inc -> /usr/local/tmp/crew/dest/usr/local/include/llvm/ProfileData/InstrProfData.inc
Fixed: /usr/local/include/llvm/ProfileData/InstrProfData.inc -> /usr/local/lib64/clang/13.0.1/include/profile/InstrProfData.inc

Attempting fix of: /usr/local/libexec/intercept-cc
 -> /usr/local/tmp/crew/dest/usr/local/libexec/intercept-c++
/usr/local/tmp/crew/llvm.20211214175054.dir/usr/local/libexec/intercept-cc
before...
lrwxrwxrwx 1 chronos chronos 56 Dec  5 23:38 /usr/local/tmp/crew/llvm.20211214175054.dir/usr/local/libexec/intercept-cc -> /usr/local/tmp/crew/dest/usr/local/libexec/intercept-c++
rm /usr/local/tmp/crew/llvm.20211214175054.dir/usr/local/libexec/intercept-cc

Exception `Errno::ENOENT' at /usr/local/lib64/ruby/3.0.0/fileutils.rb:1466 - No such file or directory @ apply2files - /usr/local/tmp/crew/llvm.20211214175054.dir/usr/local/libexec/intercept-cc

Exception `Errno::ENOENT' at /usr/local/lib64/ruby/3.0.0/fileutils.rb:800 - No such file or directory @ apply2files - /usr/local/tmp/crew/llvm.20211214175054.dir/usr/local/libexec/intercept-cc

lrwxrwxrwx 1 chronos chronos 56 Dec  5 23:38 /usr/local/tmp/crew/llvm.20211214175054.dir/usr/local/libexec/intercept-cc -> /usr/local/tmp/crew/dest/usr/local/libexec/intercept-c++
Exception `Errno::ENOENT' at /usr/local/lib64/ruby/3.0.0/fileutils.rb:1466 - No such file or directory @ apply2files - /usr/local/tmp/crew/llvm.20211214175054.dir/usr/local/libexec/intercept-cc

after...
lrwxrwxrwx 1 chronos chronos 56 Dec  5 23:38 /usr/local/tmp/crew/llvm.20211214175054.dir/usr/local/libexec/intercept-cc -> /usr/local/tmp/crew/dest/usr/local/libexec/intercept-c++
Fixed: /usr/local/libexec/intercept-c++ -> /usr/local/libexec/intercept-cc

Attempting fix of: /usr/local/lib/libscanbuild/resources/scanview.css -> /usr/local/tmp/crew/dest/usr/local/share/scan-build/scanview.css
/usr/local/tmp/crew/llvm.20211214175054.dir/usr/local/lib/libscanbuild/resources/scanview.css
before...
lrwxrwxrwx 1 chronos chronos 64 Dec  5 23:38 /usr/local/tmp/crew/llvm.20211214175054.dir/usr/local/lib/libscanbuild/resources/scanview.css -> /usr/local/tmp/crew/dest/usr/local/share/scan-build/scanview.css
rm /usr/local/tmp/crew/llvm.20211214175054.dir/usr/local/lib/libscanbuild/resources/scanview.css
Exception `Errno::ENOENT' at /usr/local/lib64/ruby/3.0.0/fileutils.rb:1466 - No such file or directory @ apply2files - /usr/local/tmp/crew/llvm.20211214175054.dir/usr/local/lib/libscanbuild/resources/scanview.css
after...
lrwxrwxrwx 1 chronos chronos 40 Dec 14 17:51 /usr/local/tmp/crew/llvm.20211214175054.dir/usr/local/lib/libscanbuild/resources/scanview.css -> /usr/local/share/scan-build/scanview.css
Fixed: /usr/local/share/scan-build/scanview.css -> /usr/local/lib/libscanbuild/resources/scanview.css
Using rdfind to convert duplicate files to hard links.
Now scanning ".", found 4335 files.
Now have 4335 files in total.
Removed 0 files due to nonunique device and inode.
Total size is 1257378136 bytes or 1 GiB
Removed 3076 files due to unique sizes from list. 1259 files left.
Now eliminating candidates based on first bytes: removed 1243 files from list. 16 files left.
Now eliminating candidates based on last bytes: removed 16 files from list. 0 files left.
Now eliminating candidates based on sha1 checksum: removed 0 files from list. 0 files left.
It seems like you have 0 files that are not unique
Totally, 0 B can be reduced.
Now making hard links.
Making 0 links.
Performing post-install for llvm...

To compile programs, use 'clang' or 'clang++'.

To avoid the repeated use of switch options,
try the wrapper scripts 'clc' or 'clc++'.

For more information, see http://llvm.org/pubs/2008-10-04-ACAT-LLVM-Intro.pdf

Llvm installed!
```
Looks like FileUtils is complaining about the broken symlink and refusing to recognize it exists? Of course this could be a container artifact...

Reply:
Any update on this PR @satmandu?

Reply:
Fixed it. This required more chomping.